ci-operator/step-registry/single-node/conf/azure/single-node-conf-azure-commands.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SINGLE_NODE_AZURE_INSTANCE_TYPE="Standard_D32s_v3"
+
+echo "Updating install-config.yaml to a single ${SINGLE_NODE_AZURE_INSTANCE_TYPE} control plane node and 0 workers"
+
+pip3 install pyyaml --user
+python3 -c '
+import yaml
+import sys
+
+input_file = sys.argv[1]
+output_file = sys.argv[2] if len(sys.argv) == 3 else sys.argv[1]  # Output in-place if destination not given
+
+with open(input_file) as f:
+    cfg = yaml.safe_load(f)
+
+# Some workflows do not define controlPlane in install-config.yaml
+if not "controlPlane" in cfg:
+  cfg["controlPlane"] = {}
+
+cfg["controlPlane"]["replicas"] = 1
+
+# Single Node Openshift requires extra memory and compute resources
+platform = cfg["controlPlane"]["platform"]
+assert "azure" in platform
+if "azure" in platform:
+    platform["azure"]["type"] = "'${SINGLE_NODE_AZURE_INSTANCE_TYPE}'"
+else:
+    raise ValueError("This step only applies to Azure, please use the correct step for your platform")
```

Comment:
nit: sorry, about this again - you forgot to remove the assertion, so this branch will never happen

Reply:
Updated (removed assertion to get a better error message)